docs/source/demos/punched_domain.geo.rst

```diff
@@ -0,0 +1,113 @@
+Creating Firedrake compatible meshes in Gmsh
```

Comment:
Grammatically, I think this wants to be "Firedrake-compatible".

Reply:
I now hyphenate this.